src/main/java/org/folio/circulation/domain/LoanRepository.java

```diff
@@ -94,6 +108,49 @@ public LoanRepository(Clients clients) {
       .thenComposeAsync(r -> r.after(this::refreshLoanRepresentation));
   }
 
+  public CompletableFuture<Result<LoanAndRelatedRecords>> replaceLoan(Loan loan) {
```

Comment:
This seems to be doing more work than replacing the loan in storage. It appears to represent a process with business logic. 

I personally don't think that belongs in a repository, and should be kept separately.